components/pagination/collections/paging_collection.js

```diff
@@ -0,0 +1,258 @@
+
+            /**
+             * Moves the collection to the next page if it exists.
+             */
+            nextPage: function () {
+                if (this.hasNextPage()) {
+                    this.setPage(this.getPageNumber() + 1);
+                }
+            },
+
+            /**
+             * Moves the collection to the previous page if it exists.
+             */
+            previousPage: function () {
+                if (this.hasPreviousPage()) {
+                    this.setPage(this.getPageNumber() - 1);
+                }
+            },
+
+            /**
+             * Adds the given field to the list of fields that can be sorted on.
+             * @param fieldName name of the field for the server API
+             * @param displayName name of the field to display to the user
+             */
+            registerSortableField: function (fieldName, displayName) {
+                this.addField(this.sortableFields, fieldName, displayName);
+            },
+
+            /**
+             * Adds the given field to the list of fields that can be filtered on.
+             * @param fieldName name of the field for the server API
+             * @param displayName name of the field to display to the user
+             */
+            registerFilterableField: function (fieldName, displayName) {
+                this.addField(this.filterableFields, fieldName, displayName);
+            },
+
+            /**
+             * For internal use only. Adds the given field to the given collection of fields.
+             * @param fields object of existing fields
+             * @param fieldName name of the field for the server API
+             * @param displayName name of the field to display to the user
+             */
+            addField: function (fields, fieldName, displayName) {
+                fields[fieldName] = {
+                    displayName: displayName
+                };
+            },
+
+            /**
+             * Returns the display name of the field that the collection is currently sorted on.
+             */
+            sortDisplayName: function () {
+                return this.sortableFields[this.sortField].displayName;
+            },
+
+            /**
+             * Returns the display name of the field that the collection is currently filtered on.
+             */
+            filterDisplayName: function () {
+                return this.filterableFields[this.filterField].displayName;
+            },
+
+            /**
+             * Sets the field to sort on. Sends a request to the server to fetch the first page of the collection with
+             * the new sort order. If successful, the collection resets to page one with the new data.
+             * @param fieldName name of the field to sort on
+             * @param toggleDirection if true, the sort direction is toggled if the given field was already set
+             */
+            setSortField: function (fieldName, toggleDirection) {
```

Comment:
The main issue I see with changing the sorting internals is that our sorting API is intentionally different from that of the paginator's. Specifically, our API has all mutating operations set the `isStale` flag, and relies on the client to call `refresh` at the appropriate time (lazy loading). The paginator, though, prefers to eagerly evaluate operations like sorting. Trying to bridge that gap would likely require either more complicated code in our collection, or a change to the API.